compiler/test/suites/types.re

```diff
@@ -366,3 +366,70 @@ describe("abstract types", ({test, testSkip}) => {
     "abc\n",
   );
 });
+
+describe("recursive types", ({test, testSkip}) => {
+  let test_or_skip =
+    Sys.backend_type == Other("js_of_ocaml") ? testSkip : test;
+
+  let assertCompileError = makeCompileErrorRunner(test);
+  let assertRun = makeRunner(test_or_skip);
+
+  assertCompileError(
+    "type_rec_incorrect_1",
+    {|
+      record Oops {
+        x: Oops
+      }
+    |},
+    "Unbound type constructor Oops. Hint: you are probably missing the `rec` keyword on this type",
```

Comment:
```suggestion
    "Unbound type constructor Oops. Are you missing the `rec` keyword on this type?",
```
What do we think about wording the error like this?

Reply:
I think I like that wording better, I just went with what I did because it mimicked the error message for not writing `rec` on mutually recursive let bindings